src/main/java/Duke.java

```diff
@@ -1,10 +1,72 @@
+import java.util.Scanner;
+import java.io.File;
+import java.io.IOException;
+
 public class Duke {
+    private TaskList tasks;
+    private Storage storage;
+    private String fileName;
+    private Parser parser;
+    private Ui ui;
+
+    public Duke(String fileName) {
+        this.fileName = fileName;
+        storage = new Storage(fileName);
+        tasks = new TaskList(storage.loadTasks());
+        parser = new Parser();
+        ui = new Ui();
+        File file = new File(this.fileName);
+
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            Ui.printLines("File creation failed.");
+        }
+    }
+    
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("src/main/data/tasks.txt").run();
+    }
+
+    private void run() {
+        ui.printIntro();
+        Scanner sc = new Scanner(System.in);
+
+        while (sc.hasNextLine()) {
+            try {
+                String input = sc.nextLine();
+                String[] split = input.split(" ");
+                String command = split[0];
+
+                if (parser.parseCommand(command, "list")) {
```

Comment:
I would suggest an alternative way to handle the different command cases using the switch statements such as 
```suggestion
                switch(parser.parseCommand(command, "list")) {
                ...
                }
```
Your code works fine though! Just another viewpoint on how you can tweak it